### DISCUSSION TITLE
Pythonic Training of Custom NER Models

### DISCUSSION THREAD

**Discussion by siamakzd** [explosion/spaCy#10440] [Help: Model Advice] [answered]
Hello,

I am new to spaCy and trying to train a custom NER model. Based on the documentations, to train a model one needs to play with some widget to generate a config and use it in command line for training.

I am wondering if there is a recipe to use spaCy like standard ML frameworks, where you can import models and create pipelines and train models (e.g. HuggingFace's way) all within Python? That can greatly improve the pipeline automation.

**Comment by polm** [comment, marked as answer]
The code for generating configs and training are all exposed as functions in spaCy. For example, see the docs for [`init config`](https://spacy.io/api/cli#init-config).
